src/app/CMakeLists.txt

```diff
@@ -514,6 +514,10 @@ if (ANDROID)
 else()
   add_executable(${QGIS_APP_NAME} MACOSX_BUNDLE WIN32 ${QGIS_APPMAIN_SRCS})
 
+  if(MSVC AND BUILD_WITH_QT6)
+    qt_disable_unicode_defines(${QGIS_APP_NAME})
+  endif(MSVC AND BUILD_WITH_QT6)
```

Comment:
Should probably not be added, https://github.com/qgis/QGIS/pull/56980 should fix this